src/app/core/components/left-menu/left-menu.component.ts

```diff
@@ -17,8 +17,24 @@ export class LeftMenuComponent {
 
   onSelectId(id: string): void {
     setTimeout(() => {
-      this.componentRef?.directiveRef?.scrollToElement(`#nav-${ id }`, -8, 300);
-    });
+      if (!this.componentRef?.directiveRef?.elementRef) {
+        return;
+      }
+
+      const elRef = <ElementRef<HTMLElement>> this.componentRef?.directiveRef?.elementRef;
+      const scrollTop: number = elRef.nativeElement.scrollTop;
+      const menuItemEl: HTMLElement | null = elRef.nativeElement.querySelector(`#nav-${ id }`);
+
+      if (!menuItemEl?.offsetTop) {
+        return;
+      }
+
+      const menuItemOffsetTop = menuItemEl.offsetTop;
+
+      if ((menuItemOffsetTop - scrollTop) <= 0) {
+        this.componentRef?.directiveRef?.scrollToElement(`#nav-${ id }`, -8, 300);
+      }
```

Comment:
After having made a remark on almost each line for useless type verification, useless optional, ... I have made a rewrite (clearer from my point of view) of exactly the same algorithm.
```suggestion
      const scrollbarDirectiveRef = this.componentRef?.directiveRef;
      if (!scrollbarDirectiveRef) return;
      const scrollbarElement = (scrollbarDirectiveRef.elementRef as ElementRef<HTMLElement>).nativeElement;

      const menuItemEl = scrollbarElement.querySelector<HTMLElement>(`#nav-${ id }`);
      if (!menuItemEl) return;

      if (menuItemEl.offsetTop - scrollbarElement.scrollTop <= 0) {
        scrollbarDirectiveRef.scrollToElement(`#nav-${ id }`, -8, 300);
      }
```